src/crypto/sha/sha512.cpp

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <libp2p/crypto/sha/sha512.hpp>
+
+#include <openssl/sha.h>
+
+namespace libp2p::crypto {
+  common::Hash512 sha512(std::string_view input) {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    const auto *bytes_ptr = reinterpret_cast<const uint8_t *>(input.data());
+    return sha512(gsl::make_span(bytes_ptr, input.length()));
+  }
+
+  common::Hash512 sha512(gsl::span<const uint8_t> input) {
+    common::Hash512 out;
+    SHA512_CTX ctx;
+    SHA512_Init(&ctx);
+    SHA512_Update(&ctx, input.data(), input.size());
```

Comment:
In this case when you use init, update, final consequently, using the doesn't seem reasonable.
Probably you should better rewrite using one function (does it exist?) or implement 3 different functions to enable continuous hash calculation. 
Not mandatory, just for your consideration.

Reply:
> using the doesn't seem
looks like the key word is omitted, pls clarify what do you mean.
The whole sha512 implementation is just a copy-paste from previously existed sha256.